src/slices/main/App.js

```diff
@@ -19,8 +18,22 @@ class App extends Component {
         onSidebarOpen: PropTypes.func.isRequired,
         onSidebarClose: PropTypes.func.isRequired,
         onCancelAuthenticate: PropTypes.func.isRequired,
+        onFinalized: PropTypes.func.isRequired,
+        setFinalizeCallback: PropTypes.func.isRequired,
+    };
+
+    state = {
+        finalize: false,
+        fakeFabScaleDirection: null,
```

Comment:
I would create a `SidebarAnimation` (or similar name) that would be responsible for all this state and synchronization. It would use a render child prop with `{ fabeIn, sidebarIn, onFabAnimationEnd, onSidebarAnimationEnd }`.

Reply:
Also the `D` is disappearing right away in the opening animation, it should scale a bit and fade out.